python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -0,0 +1,758 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sys
+import os
+import cPickle
+import sqlite3
+import json
+import subprocess
+import tempfile
+import contextlib
+import json
+import fnmatch
+import datetime
+
+import sgtk
+from sgtk.commands.clone_configuration import clone_pipeline_configuration_html
+
+###########################################################################
+# Classes
+
+class ShotgunAPI(object):
+    """
+    Public API
+    Callable methods from client. Every one of these methods can be called from the client.
+    """
+    PUBLIC_API_METHODS = [
+        "get_actions",
+        "execute_action",
+    ]
+    SYNCHRONOUS_METHODS = [
+        "get_actions",
+    ]
+
+    # Stores data persistently per wss connection.
+    WSS_KEY_CACHE = dict()
+
+    DATABASE_FORMAT_VERSION = 1
+
+    # Return codes.
+    SUCCESSFUL_LOOKUP = 0
+    UNSUPPORTED_ENTITY_TYPE = 2
+    CACHING_ERROR = 3
+    COMMAND_SUCCEEDED = 0
+    COMMAND_FAILED = 1
+
+    BASE_CONFIG_URI = os.environ.get(
```

Comment:
I have a single configuration in Shotgun that targets basic.* and uses a dev descriptor pointing to my repo on disk. It is being picked up by every project.
![screen shot 2017-04-13 at 14 45 41](https://cloud.githubusercontent.com/assets/8126447/25019240/f1ed6bae-2057-11e7-8b94-9f9804c896cb.png)